### PR TITLE
Upgrade beyla to v1.3.3

### PR DESCRIPTION
**Pull request by aabmass**
Fixes https://github.com/GoogleCloudPlatform/opentelemetry-operator-sample/issues/85

I manually verified the correct metrics are still produced.

### CODE REVIEW

File: recipes/beyla-golden-signals/beyla-daemonset.yaml [hunk 1]
```diff
@@ -41,8 +41,9 @@ spec:
               memory: 100Mi
             limits:
               memory: 500Mi
-          image: grafana/beyla:1.2.0
+          image: grafana/beyla:1.3.3
           securityContext:
+            privileged: true
```
Comment (aabmass) [inline recipes/beyla-golden-signals/beyla-daemonset.yaml:46]: Without this new line I'm seeing in the logs

`time=2024-03-05T16:20:03.490Z level=ERROR msg="couldn't trace process. Stopping process tracer" component=ebpf.ProcessTracer path=/pause pid=6748 error="attaching socket filter: operation not permitted"`

What's strange is the Beyla code hasn't changed in ~10 months https://github.com/grafana/beyla/blame/main/pkg/internal/ebpf/instrumenter.go#L211-L222

Comment (dashpole) [reply recipes/beyla-golden-signals/beyla-daemonset.yaml:46]: You can re-add the privileged flag, with a TODO above it, and I can investigate.

Comment (dashpole) [reply recipes/beyla-golden-signals/beyla-daemonset.yaml:46]: I don't want you to block on it

Comment (aabmass) [reply recipes/beyla-golden-signals/beyla-daemonset.yaml:47]: SG thanks

**Review by dashpole — APPROVED** [review]

**Pull request merged**